packages/server/src/utils.ts

```diff
@@ -12,25 +12,26 @@ export function isAsyncIterable(body: any): body is AsyncIterable<any> {
 }
 
 export interface NodeRequest {
-  protocol?: string;
-  hostname?: string;
+  protocol?: string | undefined;
+  hostname?: string | undefined;
   body?: any;
-  url?: string;
-  originalUrl?: string;
-  method?: string;
+  url?: string | undefined;
+  originalUrl?: string | undefined;
+  method: string;
   headers?: any;
-  req?: IncomingMessage | Http2ServerRequest;
-  raw?: IncomingMessage | Http2ServerRequest;
-  socket?: Socket;
+  req?: IncomingMessage | Http2ServerRequest | undefined;
+  raw?: IncomingMessage | Http2ServerRequest | undefined;
```

Comment:
This is where IncomingMessage type falls through into `const rawRequest = nodeRequest.raw || nodeRequest.req || nodeRequest`